exec/java-exec/src/main/java/org/apache/drill/exec/record/RecordBatch.java

```diff
@@ -188,6 +188,18 @@
     NOT_YET,
 
     /**
+     * Restart reading incoming batches.
+     * <p>
+     *     Prior calls to {@link #next()} returned data after all incoming batches were processed.
+     *     But with this status, need to restart reading incoming RecordBatches.
+     * </p>
+     * <p>
+     *     Currently only used internally by the Hash Aggregate.
+     * </p>
+     */
```

Comment:
Not entirely sure we want to extend the core Volcano iterator protocol to help with a single operator. In general, ALL operators should be prepared to handle this value since it is part of the public protocol.

Can we find a way to avoid this? I suspect some code restructuring, along with an operator-specific enum, would do the trick.